File: Beakjoon/_etc/Java/2193 이친수.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      long[] dp = new long[91];
      
      dp[0] = 0;
      dp[1] = 1;
      dp[2] = 1;
      
      for(int i = 3; i <= n; i++) {
          
          dp[i] = dp[i-1] + dp[i-2];
      }
      
      System.out.format("%d", dp[n]);
      
      
    }
}

// http://boj.kr/9365a86ec6b5439fb777381c20cd44c6

File: mvc1/_etc/Kotlin/2193 이친수.kt
import java.util.Scanner
import kotlin.math.*

fun main(args: Array<String>) {
    val sc: Scanner = Scanner(System.`in`)
    val n = sc.nextInt()
    var dp = LongArray(96){0}
    
    dp[0] = 0
    dp[1] = 1
    dp[2] = 1
    
    for(i in 3..n) {
        dp[i] = dp[i-2] + dp[i-1]
    }
    
    println(dp[n])
}


// http://boj.kr/cbe97cfa40bd4f6890684594b9ef3a9c
